File: content/examples/other/leetcode/problems/longest-common-subsequence/example-1/Solution.java
public class Solution {
  static int lcsNaive(String text1, String text2) {
    return dfs(text1, text2, 0, 0);
  }

  static int dfs(String a, String b, int i, int j) {
    if (i == a.length() || j == b.length()) return 0;
    if (a.charAt(i) == b.charAt(j)) return 1 + dfs(a, b, i + 1, j + 1);
    return Math.max(dfs(a, b, i + 1, j), dfs(a, b, i, j + 1));
  }

  public static void main(String[] args) {
    System.out.println(lcsNaive("abcde", "ace"));
  }
}

File: content/examples/other/leetcode/problems/longest-common-subsequence/example-1/solution.js
function lcsNaive(text1, text2) {
  function dfs(i, j) {
    if (i === text1.length || j === text2.length) return 0;
    if (text1[i] === text2[j]) return 1 + dfs(i + 1, j + 1);
    return Math.max(dfs(i + 1, j), dfs(i, j + 1));
  }
  return dfs(0, 0);
}

if (require.main === module) {
  console.log(lcsNaive("abcde", "ace"));
}

module.exports = { lcsNaive };
